server/src/test/java/org/cloudfoundry/identity/uaa/oauth/RefreshRotationTest.java

```diff
@@ -164,6 +198,35 @@ void testRefreshPublicClientWithoutRotation() {
     assertEquals("Refresh without client authentication not allowed.", exception.getMessage());
   }
 
+  @Test
+  @DisplayName("Refresh with allowpublic and rotation but existing token was not public")
+  void testRefreshPublicClientButExistingTokenWasEmptyAuthentication() {
+    BaseClientDetails clientDetails = new BaseClientDetails(tokenSupport.defaultClient);
+    clientDetails.setAutoApproveScopes(singleton("true"));
+    tokenSupport.clientDetailsService.setClientDetailsStore(IdentityZoneHolder.get().getId(), Collections.singletonMap(CLIENT_ID, clientDetails));
+    AuthorizationRequest authorizationRequest = new AuthorizationRequest(CLIENT_ID, tokenSupport.requestedAuthScopes);
+    authorizationRequest.setResourceIds(new HashSet<>(tokenSupport.resourceIds));
+    Map<String, String> azParameters = new HashMap<>(authorizationRequest.getRequestParameters());
+    azParameters.put(GRANT_TYPE, GRANT_TYPE_AUTHORIZATION_CODE);
+    authorizationRequest.setRequestParameters(azParameters);
+    authorizationRequest.setExtensions(Map.of(CLIENT_AUTH_METHOD, CLIENT_AUTH_EMPTY));
+    OAuth2Request oAuth2Request = authorizationRequest.createOAuth2Request();
+    OAuth2Authentication authentication = new OAuth2Authentication(oAuth2Request, tokenSupport.defaultUserAuthentication);
+    new IdentityZoneManagerImpl().getCurrentIdentityZone().getConfig().getTokenPolicy().setRefreshTokenRotate(true);
+    CompositeToken accessToken = (CompositeToken) tokenServices.createAccessToken(authentication);
+
+    assertThat(UaaTokenUtils.getClaims(accessToken.getValue()), hasEntry(CLIENT_AUTH_METHOD, CLIENT_AUTH_NONE));
+    String refreshTokenValue = accessToken.getRefreshToken().getValue();
+    assertThat(refreshTokenValue, is(notNullValue()));
+
+    new IdentityZoneManagerImpl().getCurrentIdentityZone().getConfig().getTokenPolicy().setRefreshTokenRotate(false);
```

Comment:
How does `tokenPolicy.refreshTokenRotate` interact with the logic added in this PR? Based on this test, it seems like when `tokenPolicy.refreshTokenRotate=false`, you forbid the public client to refresh tokens, and when `tokenPolicy.refreshTokenRotate=true`, you allow?

Why?

Reply:
This refreshTokenRotate is a prerequisite for the public usage, see my explanations below about https://oauth.net/2.1/.
Yes, I forbid the public usage without rotation because of standard . 
The refresh rotation is an important feature but I can enforce this only for new things.